shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -66,6 +68,22 @@ impl Transform for CassandraPeersRewrite {
 
         Ok(response)
     }
+
+    async fn transform_pushed<'a>(&'a mut self, mut message_wrapper: Wrapper<'a>) -> ChainResponse {
+        for message in &mut message_wrapper.messages {
+            if let Some(Frame::Cassandra(frame)) = message.frame() {
+                if let Event(ServerEvent::StatusChange(StatusChange { addr, .. })) =
+                    &mut frame.operation
+                {
+                    addr.addr.set_port(self.port);
+                    message.invalidate_cache();
+                }
+            }
+        }
+
+        let response = message_wrapper.call_next_transform_pushed().await?;
```

Comment:
What if we removed the need for this call, directly returned results in transform_pushed and had the chain manage feeding the result of one transform into the next?
My reasoning:
1. simpler API for users that better represents the flow of data, its truly one way now.
2. branching transforms with side chains wont be able to handle cassandra events anyway, so no benefit to having that flexibility here.

My only fear here is that reason 2 is incorrect and we might end up needing to handle at least some of the branching transforms. Maybe redis pubsub could support branching transforms?

So... in conclusion, unless you can see that 2 is definitely true, I'm happy to just land as is for now, we can always refactor later.

Reply:
I think we should keep it close to the original transform model for now. After we implement more we will have a better idea of what to do.

Reply:
yep, that sounds good